### PR TITLE
Added Unit Tests for TiltSwitchHelper Class

### LINKED ISSUE #301: Write Unit Tests for TiltSwitchHelper Class
**Description:**
The TiltSwitchHelper class manages the state of a tilt switch using Pi4J's DigitalInput and event listeners. We need to write unit tests to ensure that its behavior is correct and reliable, particularly regarding event listener management and state updates.

**Tasks:**

1. Test Initialization Logic:
  Verify that the isTilted field is correctly initialized based on the initial state of the DigitalInput object passed to the constructor.

2. Test Event Listener Management:
  Confirm that listeners are properly added using the addEventListener method.
  Ensure that the correct listener is invoked when the state of the tilt switch changes.
  Validate that listeners are removed using the removeEventListener method, and that the listener is set to null afterward.
3. Test State Update:
  Verify that the isTilted field accurately reflects the tilt switch's state after it changes.
  Ensure that the state of the isTilted field is updated when the listener is triggered (i.e., when the tilt switch's state changes).
4. Test Logging:
  Verify that the initialize method logs the expected message ("Initializing Tilt Switch").
5. Mocking Dependencies:
  Use a mocking framework (e.g., Mockito) to mock the DigitalInput and DigitalStateChangeListener dependencies.
6. Edge Cases:
  Test the behavior when no event listener is attached but the state of the tilt switch changes.
  Ensure that removeEventListener behaves correctly when no listener has been added.

**Acceptance Criteria**

- [ ] All test cases outlined in the tasks above are implemented.
- [x] Code coverage for the TiltSwitchHelper class is at least 70%.
- [x] Tests are written using a Java JUnit testing framework
- [x] Mocking is implemented using a framework such as Mockito or a similar tool.
- [x] All tests pass successfully, and edge cases are handled gracefully.

### PR DESCRIPTION
**Pull request by RalphTan37**
Fixes #301 

**What was changed?**

Added unit tests for the TiltSwitchHelper class using JUnit 5 and Mockito. The unit tests include checking the initialization logic, the event listener logic, the state of the mockDigitalInput when an update occurs, and the edge cases. There is no unit test for logging the initialization methods yet.

**Why was it changed?**

The unit tests were implemented to check the initialization logic, state updates, and event listener management. The TiltSwitchHelper class is used to handle the tilt switch sensor using the Pi4J Library. The unit tests ensure that the TiltSwitchHelper class functions correctly, reliably, and handles edge cases. These unit tests strengthen assurance in the codebase before hardware integration on the Raspberry Pi.

**How was it changed?**

The unit tests are implemented using JUnit 5 and Mockito to simulate the different states of the DigitalState and to check that the isTilted field is correctly initialized. Unit tests for the addEventListener and removeEventListener methods to handle a custom listener so that they can allow external components to react dynamically to tilt sensor changes. Whenever the state of mockDigitalInput changes, the unit tests validate and ensure that the TiltSwitchHelper class correctly interprets the sensor's state.

Tests

### CODE REVIEW

File: pi4micronaut-utils/src/test/java/com/opensourcewithslu/inputdevices/TiltSwitchHelperTests.java [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+        verify(mockDigitalInput, times(1)).addListener(any(DigitalStateChangeListener.class));
+    }
+
+    @Test
+    public void initializeStateLow(){
+        when(mockDigitalInput.isHigh()).thenReturn(false);
+        TiltSwitchHelper tiltSwitchHelper = new TiltSwitchHelper(mockDigitalInput);
+
+        //Verify that isTilted is initialized to false
+        assertFalse(tiltSwitchHelper.isTilted, "isTilted should be false when the input is low.");
```
Comment (avaenk) [inline pi4micronaut-utils/src/test/java/com/opensourcewithslu/inputdevices/TiltSwitchHelperTests.java:40]: Are you able to make this a variable since you use it in initialize high and low states?

**Review by yrlmanoharreddy — CHANGES_REQUESTED** [review]
The line verify(mockDigitalInput, times(1)).addListener(any(DigitalStateChangeListener.class)); appears in multiple functions, which is redundant. It would be more efficient to refactor this line into a single method to avoid repetition. Additionally, you're calling reset(mockDigitalInput) in several tests, but since you're already creating a fresh mock in @BeforeEach, resetting it may not be necessary. Removing this will simplify the code and eliminate redundancy, making it more concise and easier to maintain. Great work overall—just these small changes will improve code clarity and efficiency.

**Review by yrlmanoharreddy — APPROVED** [review]
The implementation of TiltSwitchHelperTest looks good overall. The test cases cover the expected functionality, and the mocking is properly handled, this PR is approved. Nice work!
